enumerables.rb

```diff
@@ -0,0 +1,107 @@
+# Enumerable methods
+
+module Enumerable
+  # my_each
+
+  def my_each
+    length.times { |i| yield(self[i]) }
+    self
+  end
+
+  # end of my_each
+
+  # my_each_with_index
+  def my_each_with_index
+    length.times { |i| yield(self[i], i) }
+    self
+  end
+
+  # end of my_each_with_index
+
+  # my_select
+
+  def my_select
+    arr = []
+    my_each { |i| arr.push(i) if yield(i) }
+    arr
+  end
+
+  # end of my_select
+
+  # my_all?
+
+  def my_all?
+    if_all = true
+    my_each do |i|
+      break if_all = false unless yield(i)
+    end
+    if_all
+  end
+
+  # end of my_all?
+
+  # my_any?
+
+  def my_any?
+    if_any = false
+    my_each do |i|
+      break if_any = true if yield(i)
+    end
+    if_any
+  end
+
+  # end of my_any?
+
+  # my_none?
+
+  def my_none?
+    if_none = true
+    my_each do |i|
+      break if_none = false if yield(i)
+    end
+    if_none
+  end
+
+  # end of my_none?
+
+  # my_count
+
+  def my_count(num)
+    count = 0
+    my_each { |i| count += 1 if num == i }
+    count
+  end
+
+  # end of my_count
+
+  # my_map
+
+  def my_map(num = nil)
+    arr = []
+    if block_given?
+      my_each { |i| arr.push(yield(i)) }
+    else
+      my_each { |i| arr.push(num.call(i)) }
+    end
+    arr
+  end
```

Comment:
* [ ] should have working `my_map` identical and returns the same thing as ruby's [`map`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-map). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_map.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

#### my_map 

- [ ] returns an Enumerator if no block is given
```ruby
[2,4,7,11].my_map == [2,4,7,11].map #<Enumerator: [2, 4, 7, 11]:map
```
- Yours is retrieving a NoMethodError
![image](https://user-images.githubusercontent.com/52209067/81133550-341fc480-8f0f-11ea-986b-23e3fe7a5672.png)

- Real map
![image](https://user-images.githubusercontent.com/52209067/77697646-dc3b8880-6f74-11ea-9277-4ed4a2b3af4c.png)

- [ ] in the specifications of the project in the [Odin project](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks) # 12, you were asked to modify your my_map method to take either a proc or a block. And if both a proc and a block are given, only execute the proc. The real map doesn't have this feature, but you were asked to have it so let's work on it

Given 
```ruby
my_proc = Proc.new {|num| num > 10 }
```
Result
```ruby
puts [18, 22, 5, 6] .my_map(my_proc) {|num| num < 10 }  => true true false false
```